src/main/scala/mesosphere/marathon/api/v2/GroupNormalization.scala

```diff
@@ -0,0 +1,58 @@
+package mesosphere.marathon
+package api.v2
+
+import mesosphere.marathon.state.PathId
+
+object GroupNormalization {
+
+  def partialUpdateNormalization(conf: MarathonConf): Normalization[raml.GroupPartialUpdate] = Normalization { update =>
+    update.copy(enforceRole = Some(effectiveEnforceRole(conf, update.enforceRole)))
+  }
+
+  def updateNormalization(conf: MarathonConf, id: PathId): Normalization[raml.GroupUpdate] = Normalization { update =>
+    // Only update if this is not a scale or rollback
+    if (update.version.isEmpty && update.scaleBy.isEmpty) {
```

Comment:
that means it's ok to update enforceRole if we added or removed apps or dependencies?

I think we should also verify that `update.apps.isEmpty` and `update.dependencies.isEmpty` as well.

Or can that be done somewhere while validating? How/when is the GroupPartialUpdate converted into a GroupUpdate?

Reply:
I've found only validations that `version` or `scaleBy` must be alone.

Reply:
`theOnlyDefinedOptionIn` is only used for these two.

Reply:
I think we discussed disallowing this, since one could rollback a deployment, and accidentally rolled back and enforceGroup property.

I think I would also be for disallowing the specification of enforceRole and defining new apps and services, similarly (forcing these to be two separate transactions). That would solve a known issue that we have today in which the already committed group is checked for role enforcement, while the pending transaction group is ignored. This is surprising.